drivers/wifi/winc1500/bsp/source/nm_bsp.c

```diff
@@ -0,0 +1,169 @@
+ *    and/or other materials provided with the distribution.
+ *
+ * 3. The name of Atmel may not be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ATMEL "AS IS" AND ANY EXPRESS OR IMPLIED
+ * WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT ARE
+ * EXPRESSLY AND SPECIFICALLY DISCLAIMED. IN NO EVENT SHALL ATMEL BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * \asf_license_stop
+ *
+ */
+
+#include <board.h>
+#include <gpio.h>
+
+#include "bsp/include/nm_bsp.h"
+#include "common/include/nm_common.h"
+#include "conf_winc.h"
+
+//#include <misc/printk.h>
+//#define PRINT printk
+#define PRINT(x...)
+
+/**
+ *
+ */
+sWINC1500 winc1500;
+
+static tpfNmBspIsr gpfIsr;
+
+static inline void chip_isr(struct device *port,
+			    struct gpio_callback *cb,
+			    uint32_t pins)
+{
+	if (gpfIsr) {
+		gpfIsr();
+	}
+}
+
+/*
+ *	@fn		nm_bsp_init
+ *	@brief	Initialize BSP
+ *	@return	0 in case of success and -1 in case of failure
+ */
+sint8 nm_bsp_init(void)
+{
+	PRINT("nm_bsp_init:\n");
+	gpfIsr = NULL;
+
+	/* Initialize chip IOs. */
+	winc1500.gpios = winc1500_configure_gpios();
+
+	winc1500.spi_dev = device_get_binding(CONFIG_WINC1500_SPI_DRV_NAME);
+
+	/* Perform chip reset. */
+	nm_bsp_reset();
+
+	return M2M_SUCCESS;
+}
+
+/**
+ *	@fn		nm_bsp_deinit
+ *	@brief	De-iInitialize BSP
+ *	@return	0 in case of success and -1 in case of failure
+ */
+sint8 nm_bsp_deinit(void)
+{
+	/* TODO */
+	return M2M_SUCCESS;
+}
+
+/**
+ *	@fn		nm_bsp_reset
+ *	@brief	Reset NMC1500 SoC by setting CHIP_EN and RESET_N signals low,
+ *           CHIP_EN high then RESET_N high
+ */
+void nm_bsp_reset(void)
+{
+	gpio_pin_write(winc1500.gpios[WINC1500_GPIO_IDX_CHIP_EN],
+			       CONFIG_WINC1500_GPIO_CHIP_EN, 0);
+	gpio_pin_write(winc1500.gpios[WINC1500_GPIO_IDX_RESET_N],
+			       CONFIG_WINC1500_GPIO_RESET_N, 0);
+	nm_bsp_sleep(100);
+	gpio_pin_write(winc1500.gpios[WINC1500_GPIO_IDX_CHIP_EN],
+			       CONFIG_WINC1500_GPIO_CHIP_EN, 1);
+	nm_bsp_sleep(10);
+	gpio_pin_write(winc1500.gpios[WINC1500_GPIO_IDX_RESET_N],
+			       CONFIG_WINC1500_GPIO_RESET_N, 1);
+	nm_bsp_sleep(10);
+}
+
+/*
+ *	@fn		nm_bsp_sleep
+ *	@brief	Sleep in units of mSec
+ *	@param[IN]	u32TimeMsec
+ *				Time in milliseconds
+ */
+void nm_bsp_sleep(uint32 u32TimeMsec)
+{
+	k_busy_wait(u32TimeMsec*1000);
```

Comment:
so no camelcase, but change this to:

K_MS(time_msec) * USEC_PER_SEC

USEC_PER_SEC is found in include/sys_clock.h